jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/client/GumbyClient.kt

```diff
@@ -37,64 +49,124 @@ class GumbyClient(private val project: Project) {
+            result = apiCall()
+            return result
+        } catch (e: Exception) {
+            LOG.error(e) { "$apiName failed: ${e.message}" }
+            CodetransformTelemetry.logApiError(
+                codeTransformApiErrorMessage = e.message.toString(),
+                codeTransformApiNames = apiName,
+                codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                codeTransformJobId = jobId,
+            )
+            throw e // pass along error to callee
+        } finally {
+            CodetransformTelemetry.logApiLatency(
+                codeTransformApiNames = apiName,
+                codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                codeTransformRunTimeLatency = calculateTotalLatency(startTime, Instant.now()),
+                codeTransformUploadId = uploadId,
+                codeTransformJobId = jobId,
+                codeTransformRequestId = result?.responseMetadata()?.requestId(),
+            )
+        }
     }
 
     suspend fun downloadExportResultArchive(jobId: JobId): MutableList<ByteArray> {
+        val startTime = Instant.now()
         val byteBufferList = mutableListOf<ByteArray>()
         val checksum = AtomicReference("")
-        val result = streamingBearerClient().exportResultArchive(
-            ExportResultArchiveRequest.builder()
-                .exportId(jobId.id)
-                .exportIntent(ExportIntent.TRANSFORMATION)
-                .build(),
-            ExportResultArchiveResponseHandler.builder().subscriber(
-                ExportResultArchiveResponseHandler.Visitor.builder()
-                    .onBinaryMetadataEvent {
-                        checksum.set(it.contentChecksum())
-                    }.onBinaryPayloadEvent {
-                        val payloadBytes = it.bytes().asByteArray()
-                        byteBufferList.add(payloadBytes)
-                    }.onDefault {
-                        LOG.warn { "Received unknown payload stream: $it" }
-                    }
+        try {
+            val result = streamingBearerClient().exportResultArchive(
+                ExportResultArchiveRequest.builder()
+                    .exportId(jobId.id)
+                    .exportIntent(ExportIntent.TRANSFORMATION)
+                    .build(),
+                ExportResultArchiveResponseHandler.builder().subscriber(
+                    ExportResultArchiveResponseHandler.Visitor.builder()
+                        .onBinaryMetadataEvent {
+                            checksum.set(it.contentChecksum())
+                        }.onBinaryPayloadEvent {
+                            val payloadBytes = it.bytes().asByteArray()
+                            byteBufferList.add(payloadBytes)
+                        }.onDefault {
+                            LOG.warn { "Received unknown payload stream: $it" }
+                        }
+                        .build()
+                )
```

Comment:
you can still use the fluent builder for API request object
```suggestion
        val result = streamingBearerClient().exportResultArchive(
            {
                it.exportId(jobId.id)
                it.exportIntent(ExportIntent.TRANSFORMATION)
            },
            ExportResultArchiveResponseHandler.builder().subscriber(
...
...
```

 not really seeing an obvious nice way to write the handler visitor though

Reply:
it should match the `exportResultArchive(Consumer<ExportResultArchiveRequest.Builder> exportResultArchiveRequest, ExportResultArchiveResponseHandler asyncResponseHandler)` method signature and `it` is an instance of `ExportResultArchiveRequest.Builder`